src/Engine/ProtoCore/Lang/CallSite.cs

```diff
@@ -855,13 +869,17 @@ private void UpdateCallsiteExecutionState(Object callsiteData, RuntimeCore runti
                     HashSet<FunctionEndPoint> lookups;
                     if (funcGroup.CanGetExactMatchStatics(context, reducedParams, stackFrame, runtimeCore, out lookups))
                     {
-                        //Otherwise we have a cluster of FEPs that can be used to dispatch the array
-                        resolvesFeps = new List<FunctionEndPoint>(lookups);
-                        replicationInstructions = replicationOption;
-                        return;
+                        if (replicationInstructions == null || IsCompatibleReplicationOption(replicationInstructions, replicationOption))
+                        {
+                            //Otherwise we have a cluster of FEPs that can be used to dispatch the array
```

Comment:
I find this comment confusing - what is the `otherwise` referring to - theres no comment before this and we are inside an if statement... not an else statement -